File: r/src/operators.R
# Function to demonstrate operators in R
demonstrate_operators <- function() {
  cat("=== Operators ===\n")
  
  # Arithmetic operators
  cat("Arithmetic Operators:\n")
  a <- 10
  b <- 3
  
  cat("a =", a, ", b =", b, "\n")
  cat("Addition (a + b):", a + b, "\n")
  cat("Subtraction (a - b):", a - b, "\n")
  cat("Multiplication (a * b):", a * b, "\n")
  cat("Division (a / b):", a / b, "\n")
  cat("Integer Division (a %/% b):", a %/% b, "\n")
  cat("Modulus (a %% b):", a %% b, "\n")
  cat("Exponentiation (a ^ b):", a ^ b, "\n")
  
  # Comparison operators
  cat("\nComparison Operators:\n")
  cat("Equal to (a == b):", a == b, "\n")
  cat("Not equal to (a != b):", a != b, "\n")
  cat("Greater than (a > b):", a > b, "\n")
  cat("Less than (a < b):", a < b, "\n")
  cat("Greater than or equal to (a >= b):", a >= b, "\n")
  cat("Less than or equal to (a <= b):", a <= b, "\n")
  
  # Logical operators
  cat("\nLogical Operators:\n")
  x <- TRUE
  y <- FALSE
  
  cat("x =", x, ", y =", y, "\n")
  cat("AND (x & y):", x & y, "\n")
  cat("OR (x | y):", x | y, "\n")
  cat("NOT !x:", !x, "\n")
  
  # Assignment operators
  cat("\nAssignment Operators:\n")
  val <- 5
  cat("Standard assignment (<-):", val, "\n")
  
  val <<- 6  # Global assignment
  cat("Global assignment (<<-):", val, "\n")
  
  val = 7    # Alternative assignment
  cat("Alternative assignment (=):", val, "\n")
  
  # Special operators
  cat("\nSpecial Operators:\n")
  vec <- 1:5
  cat("Sequence operator (1:5):", vec, "\n")
  
  mat <- matrix(1:9, 3, 3)
  cat("Matrix multiplication operator (%*%):\n")
  print(mat %*% mat)
  
  cat("\n")
}

File: ruby/lib/operators.rb
module Operators
  def self.demonstrate
    puts "\n=== Operators ==="
    
    a = 10
    b = 3
    
    # Arithmetic operators
    puts "Arithmetic Operators:"
    puts "Addition: #{a + b}"
    puts "Subtraction: #{a - b}"
    puts "Multiplication: #{a * b}"
    puts "Division: #{a / b}"       # Integer division
    puts "Float Division: #{a.to_f / b}"
    puts "Modulus: #{a % b}"
    puts "Exponent: #{a ** 2}"     # Power
    
    # Comparison operators
    puts "\nComparison Operators:"
    puts "Equal: #{a == b}"
    puts "Not Equal: #{a != b}"
    puts "Greater Than: #{a > b}"
    puts "Less Than: #{a < b}"
    puts "Greater or Equal: #{a >= b}"
    puts "Less or Equal: #{a <= b}"
    puts "Spaceship: #{a <=> b}"   # Returns -1, 0, or 1
    
    # Logical operators
    puts "\nLogical Operators:"
    puts "AND: #{true && false}"
    puts "OR: #{true || false}"
    puts "NOT: #{!true}"
    
    # Ruby special operators
    puts "\nRuby Special Operators:"
    str1 = "Hello"
    str2 = "Hello"
    puts "Equal Object? #{str1.equal?(str2)}"    # Compare object identity
    puts "Equal Value? #{str1 == str2}"          # Compare value
    puts "Pattern Match? #{'123' =~ /\d+/}"      # Regular expression match
    
    # Assignment operators
    puts "\nAssignment Operators:"
    x = 5
    x += 3  # Same as x = x + 3
    puts "After += 3: #{x}"
    x -= 2
    puts "After -= 2: #{x}"
    x *= 4
    puts "After *= 4: #{x}"
  end
end